### ISSUE TITLE
Custom image filters to help labeling

### ISSUE THREAD

**Issue by twsl** [heartexlabs/label-studio#1425]
**Is your feature request related to a problem? Please describe.**
Sometimes labeling requires a great level of expertise. Additional information could be helpful, especially when working with images.

**Describe the solution you'd like**
I'd like to apply filters to help me detect objects in high resolution images. Using a framework such as [MarvinJ](http://www.marvinj.org/en/index.html) would enable filters like [gradient](https://en.wikipedia.org/wiki/Image_gradient)/[emboss](https://en.wikipedia.org/wiki/Image_embossing), threshold based black and white, or [edge detection](https://de.wikipedia.org/wiki/Sobel-Operator) and could help with the annotation. A plugin based implementation would enable users to implement their own filters. An interactive way is required to select the threshold or direction of the gradient.

**Describe alternatives you've considered**
I considered implementing it in a fork, but the web project structure in two projects is kinda confusing.

**Additional context**
Happy to help, as long as the PR is merged faster than my previous contribution.

**Comment by makseq** [comment]
@twsl MarvinJ is really cool tool! Now we are going to release interactive pre-annotations, they allow ML to assist users in the labeling, hope it helps you too!

**Comment by twsl** [comment]
In my specific usecase I need an interactive gradient via sobel filters where you can select orientation and strength to find edges and other features in greyscale images that are useful hints for experts.
Are these pre-annotations customizable? In that case I could simply replace the ML algo with a simple filter with default values

**Comment by makseq** [comment]
Yes, interactive pre-annotations are an extension of ML backend. It's a good idea to replace images from ML backend "on the fly", but our interactive annotations don't have this functionality yet.

**Comment by twsl** [comment]
Thats what I thought. But I think js based filters could offer a great benefit to many users working with greyscale/depth images where changes are subtle and hard to spot

**Comment by suahelen** [comment]
I would like to reignite this idea here. In my opinion it would be a great addition to help any Labeler do a better job. I have often come across images that were hard to label as normal RGB images, but a simple contrast enhancement did wonders. Although my interest is mainly for Image based annotations, I could also imagine that similar filters could help with other types of annotations such as timeseries data.
 Would it be an option to include this feature into your roadmap?

**Comment by makseq** [comment]
@suahelen Thank you for your feedback. If there are many requests about it, we try to implement/include PRs to our releases. So, it depends on demand.

**Comment by tanglef** [comment]
Hello, I'm upping this Feature Request as this would greatly help to show side by side / overlap with a button the original image with a processed image.
In the mean time, is there any way to show two images side by side, one original and next the processed version (with the annotation showing on both for object detection for example) ?

**Comment by AbubakarSaad** [comment]
Hello @tanglef,

Yes, that is possible. You can do something like this, which will show side by side image to annotate. 
```
<View style="display: flex; gap: 12px;">
  <View style="flex: 1;">
    <Header value="Original"/>
    <Image name="img_orig" value="$image_orig" zoom="true" zoomControl="true"/>
  </View>

  <View style="flex: 1;">
    <Header value="Processed"/>
    <Image name="img_proc" value="$image_proc" zoom="true" zoomControl="true"/>
  </View>

  <RectangleLabels name="bbox" toName="img_orig" strokeWidth="2">
    <Label value="Object" background="#ff5252"/>
  </RectangleLabels>
</View>
```
